aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -332,6 +407,261 @@ def _build_middleware_stack(self, router_middlewares: List[Callable[..., Any]])
+
+    def _get_openapi_path(
+        self,
+        *,
+        dependant: "Dependant",
+        operation_ids: Set[str],
+        model_name_map: Dict["TypeModelOrEnum", str],
+        field_mapping: Dict[Tuple["ModelField", Literal["validation", "serialization"]], "JsonSchemaValue"],
+    ) -> Tuple[Dict[str, Any], Dict[str, Any]]:
+        from aws_lambda_powertools.event_handler.openapi.dependant import get_flat_params
+
+        path = {}
+        definitions: Dict[str, Any] = {}
+
+        # Gather all the route parameters
+        operation = self._openapi_operation_metadata(operation_ids=operation_ids)
+        parameters: List[Dict[str, Any]] = []
+        all_route_params = get_flat_params(dependant)
+        operation_params = self._openapi_operation_parameters(
+            all_route_params=all_route_params,
+            model_name_map=model_name_map,
+            field_mapping=field_mapping,
+        )
+        parameters.extend(operation_params)
+
+        # Add the parameters to the OpenAPI operation
+        if parameters:
+            all_parameters = {(param["in"], param["name"]): param for param in parameters}
+            required_parameters = {(param["in"], param["name"]): param for param in parameters if param.get("required")}
+            all_parameters.update(required_parameters)
+            operation["parameters"] = list(all_parameters.values())
+
+        # Add the request body to the OpenAPI operation, if applicable
+        if self.method.upper() in METHODS_WITH_BODY:
+            request_body_oai = self._openapi_operation_request_body(
+                body_field=self.body_field,
+                model_name_map=model_name_map,
+                field_mapping=field_mapping,
+            )
+            if request_body_oai:
+                operation["requestBody"] = request_body_oai
+
+        # Add the response to the OpenAPI operation
+        responses = operation.setdefault("responses", {})
+        success_response = responses.setdefault("200", {})
+        success_response["description"] = self.response_description or _DEFAULT_OPENAPI_RESPONSE_DESCRIPTION
+        success_response["content"] = {"application/json": {"schema": {}}}
+        json_response = success_response["content"].setdefault("application/json", {})
+
+        # Add the response schema to the OpenAPI response
+        json_response.update(
+            self._openapi_operation_return(
+                operation_id=self.operation_id,
+                param=dependant.return_param,
+                model_name_map=model_name_map,
+                field_mapping=field_mapping,
+            ),
+        )
+
+        # Add validation responses
+        operation["responses"]["422"] = {
+            "description": "Validation Error",
+            "content": {
+                "application/json": {
+                    "schema": {"$ref": COMPONENT_REF_PREFIX + "HTTPValidationError"},
+                },
+            },
+        }
```

Comment:
Should this be optional or mandatory? I don't have a strong opinion, but I don't think this should be mandatory with HTTP 422. IMO customers may want to change this to 400 or another HTTP error code.

Reply:
This is in line with FastAPI. As we are doing validation of the data, there's always the possibility of the data not matching the content. As such, clients must be prepared to handle the [422](https://http.dev/422) that our resolver will return in that scenario.